code/game/objects/items/weapons/storage/lockbox.dm

```diff
@@ -107,51 +118,67 @@
 	icon_broken = "medalbox+b"
 	can_hold = list(/obj/item/clothing/accessory/medal)
 
+/obj/item/weapon/storage/lockbox/medal/AltClick()
+	if(!locked)
+		open = (open ? FALSE : TRUE)
+		update_icon()
+	..()
+
 /obj/item/weapon/storage/lockbox/medal/PopulateContents()
+	new /obj/item/clothing/accessory/medal/gold/captain(src)
+	new /obj/item/clothing/accessory/medal/silver/valor(src)
 	new /obj/item/clothing/accessory/medal/silver/valor(src)
+	new /obj/item/clothing/accessory/medal/silver/security(src)
 	new /obj/item/clothing/accessory/medal/bronze_heart(src)
+	new /obj/item/clothing/accessory/medal/plasma/nobel_science(src)
+	new /obj/item/clothing/accessory/medal/plasma/nobel_science(src)
 	for(var/i in 1 to 3)
 		new /obj/item/clothing/accessory/medal/conduct(src)
-	new /obj/item/clothing/accessory/medal/gold/captain(src)
-	new /obj/item/clothing/accessory/medal/silver/security(src)
-	new /obj/item/clothing/accessory/medal/plasma(src)
-	new /obj/item/clothing/accessory/medal/plasma/nobel_science(src)
-	new /obj/item/clothing/accessory/medal/gold/heroism(src)
 
-/obj/item/weapon/storage/lockbox/secmedal
+/obj/item/weapon/storage/lockbox/medal/update_icon()
+	cut_overlays()
+	if(locked)
+		icon_state = "medalbox+l"
+		open = FALSE
+	else
+		icon_state = "medalbox"
+		if(open)
+			icon_state += "open"
+		if(broken)
+			icon_state += "+b"
+		if(contents && open)
+			for (var/i in 1 to contents.len)
+				var/medaltype = "medal"
+				var/obj/item/clothing/accessory/medal/M = contents[i]
+				if(istype(M, /obj/item/clothing/accessory/medal/gold))

```

Comment:
Given that you were able to perfectly work out what I mean, perhaps what I wrote really was enough, meaning you've just defeated your own point.

I don't explain things like this when I've seen the PR author produce much better code, this is something they almost certainly know how to do, but simply didn't, the comment will draw their attention, they'll re-evaluate what they've done, and be like "Oh, derp"